libraries/rush-lib/src/api/CustomTipsConfiguration.ts

```diff
@@ -56,12 +56,44 @@ export interface ICustomTipItemJson {
  *
  * @beta
  */
-export type CustomTipId =
-  // The rush.json "ensureConsistentVersions" validation.
-  | 'TIP_RUSH_INCONSISTENT_VERSIONS'
-  // In the future, plugins can contribute other strings.
-  | string;
+export enum CustomTipIdEnum {
+  'TIP_RUSH_INCONSISTENT_VERSIONS' = 'TIP_RUSH_INCONSISTENT_VERSIONS',
+  'TIP_PNPM_NO_MATCHING_VERSION' = 'TIP_PNPM_NO_MATCHING_VERSION'
+}
+
+enum CustomTipSeverity {
+  'Warning',
+  'Error',
+  'Info'
+}
+
+export enum CustomTipType {
+  'rush',
+  'pnpm'
+}
+
+export interface ITipMetadata {
+  tipId: CustomTipIdEnum;
+  severity: CustomTipSeverity;
+  type: CustomTipType;
+  isMatch?: (str: string) => boolean;
```

Comment:
Can you include a documentation comment for this property?

Reply:
Added. Please take a look at whether it's too much detail :D